moonstreamapi/moonstreamapi/routes/queries.py

```diff
@@ -263,36 +274,34 @@ async def get_query_handler(
             )
 
         try:
-            entries = bc.search(
+            entry = bc.get_entry(
                 token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
                 journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
-                query=f"tag:approved tag:query_id:{query_id} !tag:preapprove",
-                limit=1,
-                timeout=5,
+                entry_id=query_id,
             )
         except BugoutResponseException as e:
             logger.error(f"Error in get query: {str(e)}")
             raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
         except Exception as e:
             raise MoonstreamHTTPException(status_code=500, internal_error=e)
 
-        if len(entries.results) == 0:
-            raise MoonstreamHTTPException(
-                status_code=403, detail="Query not approved yet."
-            )
     else:
         entries_results = cast(List[BugoutSearchResult], entries.results)
         query_id = entries_results[0].entry_url.split("/")[-1]
+        entry = entries_results[0]

```

Comment:
I think this structure is wrong, what if we `try: entries = bc.search(` (line 252) failed, then we fetch line 277 and get entry, so our `entries is None` and we will fail when will try to get `entries.results` at line 289

Reply:
`try: entries = bc.search( (line 252) failed` Give you raise MoonstreamHTTPException next code will be exacute only if search is finished.

Reply:
But it return empty list, then `entries_results[0]` will fail